geopandas/sindex.py

```diff
@@ -1,25 +1,235 @@
-from geopandas import base
+from warnings import warn
+from collections import namedtuple
 
-if base.HAS_SINDEX:
-    from rtree.index import Index as RTreeIndex
+from pandas import notnull
+import numpy as np
 
+from . import _compat as compat
 
-class SpatialIndex(RTreeIndex):
+
+def has_sindex():
+    """
+    Dynamically checks for ability to generate spatial index.
+    """
+    return get_sindex_class() is not None
+
+
+def get_sindex_class():
     """
-    A simple wrapper around rtree's RTree Index
+    Dynamically chooses a spatial indexing backend.
+    Required to comply with _compat.USE_PYGEOS.
+    The order of preference goes PyGeos > RTree > None.
     """
+    if compat.USE_PYGEOS:
+        return PyGEOSSTRTreeIndex
+    elif compat.HAS_RTREE:
+        return RTreeIndex
+    else:
+        warn("Cannot generate spatial index: Missing package `rtree` or `pygeos`.")
+        return None
+
+
+if compat.HAS_RTREE:
+
+    from rtree.index import Index as ToblerityRTreeIndex  # noqa
```

Comment:
```suggestion
    import rtree.index
```

(then the "ToblerityRtreeIndex" aliasing is not needed)